src/Lecture3.hs

```diff
@@ -122,10 +130,10 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+  (<>) (Reward g1 prize1) (Reward g2 prize2) = Reward (g1 <> g2) (prize1 || prize2)
 
 instance Monoid Reward where
-
+  mempty = Reward (Gold 0) False
```

Comment:
You can use `mempty` for `Gold` here because it already has a `Monoid` instance 🏅 
```suggestion
  mempty = Reward mempty False
```